### PR TITLE
Support JDK 11 and JDK 17 Compilation

### PR DESCRIPTION
**Pull request by YogyaGamage**
### Purpose
Update mockito 1 to mockito 2 to support JDK 11 and JDK 17 compilation.

### When should this PR be merged
After this PR is merged to the main branch, we can't build the branch on JDK 8

### Related Issue
https://github.com/wso2/product-is/issues/14224

### CODE REVIEW

**Comment by janakamarasena** [comment]
Shouldn't the maven-compiler-plugin source and target versions change too?

File: components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/SAMLSSOServiceTest.java [hunk 1]
```diff
@@ -126,12 +125,14 @@ public void testValidateSPInitSSORequestAuthentication(String encodedAuthnReques
...
         mockserviceProviderConfigs.setDoValidateSignatureInRequests(false);
+        mockserviceProviderConfigs.setTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
         List<String> acsUrls = new ArrayList<>();
         acsUrls.add(TestConstants.ACS_URL);
         acsUrls.add(TestConstants.RETURN_TO_URL);
         mockserviceProviderConfigs.setAssertionConsumerUrls(acsUrls);
         mockStatic(SAMLSSOUtil.class);
-        when(SAMLSSOUtil.getSPInitSSOAuthnRequestValidator(any(AuthnRequest.class), any(String.class))).thenCallRealMethod();
+        when(SAMLSSOUtil.getTenantDomainFromThreadLocal()).thenReturn(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+        when(SAMLSSOUtil.getSPInitSSOAuthnRequestValidator(any(AuthnRequest.class), nullable(String.class))).thenCallRealMethod();
```
Comment (YogyaGamage) [inline components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/SAMLSSOServiceTest.java:135]: Use a dataprovider and check for null and not null cases.

File: components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/SAMLSSOServiceTest.java [hunk 2]
```diff
@@ -157,9 +158,10 @@ public void testValidateSPInitSSORequestLogout(String encodedLogoutRequest, Stri
         SAMLSSOUtil.doBootstrap();
 
         SPInitLogoutRequestProcessor spInitLogoutRequestProcessor = mock(SPInitLogoutRequestProcessor.class);
-        when(spInitLogoutRequestProcessor.process(any(LogoutRequest.class), anyString(), anyString(), anyString())).thenReturn(
-                mockValidSPInitLogoutRequestProcessing(TestConstants.ACS_URL));
+        when(spInitLogoutRequestProcessor.process(any(LogoutRequest.class), anyString(), nullable(String.class),
```
Comment (YogyaGamage) [inline components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/SAMLSSOServiceTest.java:161]: Use a dataprovider and check for null and not null cases.

File: components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/SAMLSSOServiceTest.java [hunk 3]
```diff
@@ -201,8 +203,9 @@ public void testValidateIdPInitSSORequestAuthentication() throws Exception {
         boolean isLogout = false;
 
         mockStatic(SAMLSSOUtil.class);
+        when(SAMLSSOUtil.getTenantDomainFromThreadLocal()).thenReturn(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
         when(SAMLSSOUtil.resolveIssuerQualifier(any(QueryParamDTO[].class), anyString())).thenCallRealMethod();
-        when(SAMLSSOUtil.getIdPInitSSOAuthnRequestValidator(any(QueryParamDTO[].class), anyString()))
+        when(SAMLSSOUtil.getIdPInitSSOAuthnRequestValidator(any(QueryParamDTO[].class), nullable(String.class)))
```
Comment (YogyaGamage) [inline components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/SAMLSSOServiceTest.java:208]: Use relaystate directly

File: components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/util/AssertionBuildingTest.java [hunk 4]
```diff
@@ -364,7 +376,7 @@ private void prepareIdentityPersistentManager(String attrConsumerIndex, String i
         samlssoServiceProviderDO.setEnableAttributesByDefault(true);
         samlssoServiceProviderDO.setIssuer(issuer);
         samlssoServiceProviderDO.setAssertionConsumerUrls(acsList);
-        when(identityPersistenceManager.getServiceProvider(any(Registry.class), anyString()))
+        when(identityPersistenceManager.getServiceProvider(nullable(Registry.class), anyString()))
```
Comment (YogyaGamage) [inline components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/util/AssertionBuildingTest.java:379]: Change anyString to issuer.

File: components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtilMarshallTest.java [hunk 5]
```diff
@@ -50,9 +52,9 @@ public void testUnMarshallRandomString() throws Exception {
     @Test
     public void testMarshall() throws Exception {
 
-        assertEquals(SAMLSSOUtil.marshall(SAMLSSOUtil.unmarshall(TestConstants.DECODED_POST_LOGOUT_REQUEST)),
-                TestConstants.DECODED_POST_LOGOUT_REQUEST,
-                "Marshaled Post Authentication Request is not as expected.");
+        Diff diff = new Diff(SAMLSSOUtil.marshall(SAMLSSOUtil.unmarshall(TestConstants.DECODED_POST_LOGOUT_REQUEST)),
```
Comment (YogyaGamage) [inline components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtilMarshallTest.java:55]: Add the change in the XML message after framework update

File: pom.xml [hunk 6]
```diff
@@ -407,9 +425,6 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
-                <configuration>
-                    <source>1.8</source>
```
Comment (YogyaGamage) [inline pom.xml:411]: use 11

**Pull request closed without merging**